File: Javascript/0167_Two_Sum_II_Input_Array_Is_Sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let l=0;
    let r=numbers.length-1;
    while(l<r) {
        if(numbers[l] + numbers[r] > target) {
            r--;
        }
        else if(numbers[l] + numbers[r] < target) {
            l++;
        }
        else {
            return [l+1, r+1];
        }
    }
};

File: Python/0167_Two_Sum_II_Input_Array_Is_Sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        # Time Complexity: O(N)
        # Space Complexity: O(1)
        l, r = 0, len(numbers)-1
        while l < r:
            if numbers[r] + numbers[l] > target:
                r -= 1
            elif numbers[r] + numbers[l] < target:
                l += 1
            else:
                return l+1, r+1
            